bento_wes/backends/wes_backend.py

```diff
@@ -43,44 +50,40 @@
 class WESBackend(ABC):
     def __init__(
         self,
-        tmp_dir: Path,
-        data_dir: Path,
-        workflow_timeout: int,  # Workflow timeout, in seconds
-        logger=None,
-        event_bus: EventBus | None = None,
-        workflow_host_allow_list: set | None = None,
-        bento_url: str | None = None,
-        validate_ssl: bool = True,
-        debug: bool = False,
+        settings: Settings,
+        logger: Logger,
+        event_bus: EventBus,
     ):
-        self._workflow_timeout: int = workflow_timeout
-
-        self.db: Database = get_db()
+        self.settings = settings
+        self._workflow_timeout: int = int(settings.workflow_timeout.total_seconds())
 
-        self.tmp_dir: Path = tmp_dir
-        self.data_dir: Path = data_dir
+        self.tmp_dir: Path = settings.service_temp
+        self.data_dir: Path = settings.service_data
 
-        self.output_dir: Path = data_dir / "output"  # For persistent file artifacts from workflows
+        self.output_dir: Path = self.data_dir / "output"  # For persistent file artifacts from workflows
         self.output_dir.mkdir(parents=True, exist_ok=True)
 
         self.logger = logger
-        self.event_bus = event_bus  # TODO: New event bus?
+        self.event_bus = event_bus
 
-        self.workflow_host_allow_list = workflow_host_allow_list
+        self._db_gen = get_db_with_event_bus(self.logger, self.event_bus)
+        self.db: Database = next(self._db_gen)
+
+        self.workflow_host_allow_list = parse_workflow_host_allow_list(settings.workflow_host_allow_list)
 
         # Bento-specific parameters
-        self.bento_url: str = bento_url
+        self.bento_url = str(settings.bento_url)
 
-        self.validate_ssl: bool = validate_ssl
-        self.debug: bool = debug
+        self.validate_ssl: bool = settings.bento_validate_ssl
+        self.debug: bool = settings.bento_debug
 
         self._workflow_manager: WorkflowManager = WorkflowManager(
```

Comment:
workflowmanager should be rewritten as a dependency of settings etc since it naturally is structured similar to one.